File: PolishPizza/admin/src/service/pizzaService.ts
import { api as axiosInstance } from '../api/api';
import type { PizzaRequest, PizzaResponse } from '../types/menu';

export const PizzaService = {
    
    getAll: async (): Promise<PizzaResponse[]> => {
        const response = await axiosInstance.get<PizzaResponse[]>('/menu/pizzas'); 
        return response.data;
    },

    getOne: async (id: string): Promise<PizzaResponse | null> => {
        const response = await axiosInstance.get<PizzaResponse>(`/menu/pizzas/${id}`);
        return response.data;
    },
    
    update: async (id: string, data: PizzaRequest) => {
        const response = await axiosInstance.put(`/menu/pizzas/${id}`, data);
        return response.data;
    },
    
    create: async (data: PizzaRequest) => {
        const response = await axiosInstance.post(`/menu/pizzas`, data);
        return response.data;
    },

    delete: async(id: string): Promise<void> => {
        const response = await axiosInstance.delete(`/menu/pizzas/${id}`);
        return response.data;
    }
};

File: PolishPizza/menu/src/main/java/pizzeria/menu/pizza/service/PizzaService.java
package pizzeria.menu.pizza.service;

import pizzeria.menu.pizza.dto.response.PizzaResponse;
import pizzeria.menu.pizza.model.PizzaIngredient;
import pizzeria.menu.pizza.dto.request.PizzaPatchRequest;
import pizzeria.menu.pizza.dto.request.PizzaRequest;

import java.util.List;
import java.util.UUID;

public interface PizzaService {
    List<PizzaResponse> getAllPizzas();

    PizzaResponse getPizzaById(UUID id);

    PizzaResponse save(PizzaRequest request);

    void delete(UUID id);

    PizzaResponse update(UUID id, PizzaRequest request);

    PizzaResponse patch(UUID id, PizzaPatchRequest request);

    List<PizzaIngredient> getAllPizzasIngredients(UUID pizzaId);
}
